src/command/process_update_task.rs

```diff
@@ -180,6 +202,132 @@ impl UpdateProcessor {
         }
     }
 
+    async fn not_valid_offset_message(&self) -> Result<Option<ScheduleWeatherTask>, BotError> {
+        self.cancel(Some(
+            "That's not a valid offset, it has to be a number in range [-11, 12].\n
+            If your timezone is UTC + 2 put 2, if you have UTC - 10 put -10, 0 if you have UTC timezone.\n
+            The command was cancelled"
+            .to_string(),
+        ))
+        .await?;
+
+        Ok(None)
+    }
+
+    async fn process_offset(&self) -> Result<Option<ScheduleWeatherTask>, BotError> {
+        match self.text.parse::<i8>() {
+            Ok(number) => {
+                if !(-11..=12).contains(&number) {
+                    return self.not_valid_offset_message().await;
+                }
+
+                let offset = number;
+                // we have hour and minutes well formatted stored in selected see in process_time func.
+                let vec: Vec<&str> = self.chat.selected.as_ref().unwrap().split(':').collect();
+                // This unwraps are safe because we know that is stored in the correct format.
+                let user_hour = vec[0].parse::<i8>().unwrap();
+                let minutes = vec[1].parse::<i8>().unwrap();
+
+                let schedule_weather_task = ScheduleWeatherTask::builder()
+                    .user_hour(user_hour)
+                    .minutes(minutes)
+                    .offset(offset)
+                    .username(self.username.clone())
+                    .chat_id(self.chat.id)
+                    // Safe unwrap checked in process_time func.
+                    .default_city_id(self.chat.default_city_id.unwrap())
+                    .build();
+
+                self.return_to_initial().await?;
+
+                let minutes_pretty = if minutes < 10 {
+                    format!("0{}", vec[1])
+                } else {
+                    vec[1].to_string()
+                };
+
+                let text = format!(
+                    "Weather info scheduled every day at {}:{} UTC {}",
+                    vec[0], minutes_pretty, offset
+                );
+
+                self.send_message(&text).await?;
+
+                Ok(Some(schedule_weather_task))
+            }
+
+            Err(_) => self.not_valid_offset_message().await,
+        }
+    }
+
+    async fn not_time_message(&self) -> Result<(), BotError> {
+        self.cancel(Some(
+            "That's not a well formatted time, it has to be formatted with this format `hour:minutes` being hour a number in range [0,23] 
+            and minutes a number in range [0,59]. The command was cancelled"
+            .to_string(),
+        ))
+        .await
+    }
+
+    fn parse_time(hour_or_minutes: &str, max_range: i32, min_range: i32) -> i32 {
+        match hour_or_minutes.parse::<i32>() {
+            Ok(number) => {
+                if !(min_range..=max_range).contains(&number) {
+                    -1
```

Comment:
instead of returning -1, use can just use `Result` enum

Reply:
[parse_time func with a Result](https://github.com/pxp9/weather_bot_rust/pull/29/commits/e3c7e1f14212ab3e599e5ad38ef177b6857805da)